File: meituan/p1088.cc
#include <bits/stdc++.h>
using namespace std;

int solve(vector<int> &colors, int k) {
  int ans = 0;
  int n = colors.size();
  unordered_map<int, int> cnt;

  int left = 0, right = 0;
  while (right < n) {
    int cr = colors[right++];
    if (cnt.find(cr) != cnt.end()) {
      cnt[cr]++;
    } else {
      cnt[cr] = 1;
    }
    while (cnt.size() > k) {
      int cl = colors[left++];
      cnt[cl] -= 1;
      if (cnt[cl] == 0) {
        cnt.erase(cl);
      }
    }
    ans = max(ans, right - left);
  }
  return ans;
}

int main(int argc, char *argv[]) {
  int n, k;
  cin >> n >> k;
  vector<int> colors(n);
  for (int i = 0; i < n; i++) {
    cin >> colors[i];
  }

  auto res = solve(colors, k);
  cout << res << endl;
  return 0;
}

File: meituan/p1088.py
from typing import List


def solve(colors: List[int], k: int) -> int:
  n = len(colors)
  ans = 0
  cnt = {}

  left, right = 0, 0
  while right < n:
    cr = colors[right]
    right += 1
    cnt[cr] = cnt.get(cr, 0) + 1 

    while len(cnt) > k:
      cl = colors[left]
      left += 1
      cnt[cl] -= 1
      if cnt[cl] == 0:
        del cnt[cl]
    
    ans = max(ans, right - left)
  return ans

n, k = map(int, input().split())
colors = list(map(int, input().split()))

ans = solve(colors, k)
print(ans)